packages/venia-concept/src/RootComponents/Category/category.js

```diff
@@ -1,119 +1,97 @@
-import React, { Component } from 'react';
+import React, { useEffect } from 'react';
 import { string, number, shape } from 'prop-types';
-import { compose } from 'redux';
-import { connect, Query } from 'src/drivers';
+import { useQuery } from '@magento/peregrine';
 
-import classify from 'src/classify';
-import { setCurrentPage, setPrevPageTotal } from 'src/actions/catalog';
-import { loadingIndicator } from 'src/components/LoadingIndicator';
+import { mergeClasses } from 'src/classify';
+import categoryQuery from 'src/queries/getCategory.graphql';
 import CategoryContent from './categoryContent';
+import { loadingIndicator } from 'src/components/LoadingIndicator';
 import defaultClasses from './category.css';
-import categoryQuery from 'src/queries/getCategory.graphql';
 
-class Category extends Component {
-    static propTypes = {
-        id: number,
-        classes: shape({
-            gallery: string,
-            root: string,
-            title: string
-        }),
-        currentPage: number,
-        pageSize: number,
-        prevPageTotal: number
-    };
+const Category = props => {
+    const {
+        id,
+        currentPage,
+        pageSize,
+        prevPageTotal,
+        setCurrentPage,
+        setPrevPageTotal
+    } = props;
 
-    // TODO: Should not be a default here, we just don't have
-    // the wiring in place to map route info down the tree (yet)
-    static defaultProps = {
-        id: 3
+    const pageControl = {
+        currentPage: currentPage,
+        setPage: setCurrentPage,
+        updateTotalPages: setPrevPageTotal,
+        totalPages: prevPageTotal
     };
 
-    componentDidUpdate(prevProps) {
-        // If the current page has changed, scroll back up to the top.
-        if (this.props.currentPage !== prevProps.currentPage) {
-            window.scrollTo(0, 0);
-        }
-    }
-
-    render() {
-        const {
-            id,
-            classes,
-            currentPage,
-            pageSize,
-            prevPageTotal,
-            setCurrentPage,
-            setPrevPageTotal
-        } = this.props;
+    const [queryResult, queryApi] = useQuery(categoryQuery);
+    const { data, error, loading } = queryResult;
+    const { runQuery, setLoading } = queryApi;
+    const classes = mergeClasses(defaultClasses, props.classes);
```

Comment:
As we move to functional components we should colocate the variables with their uses, IMO (that may be controversial). But I think this `classes` assignment can live right above the `return` where it is used. May as well go through and make sure other variables are assigned right before use as well.